dev/io.openliberty.security.jakartasec.3.0.internal_fat.refresh/fat/src/io/openliberty/security/jakartasec/fat/refresh/tests/BasicRefreshTests.java

```diff
@@ -0,0 +1,865 @@
+
+    }
+
+    /**
+     * Build the long app name based on the different test settings passed in and use that to obtain the mapped app name which is much shorter, but not very descriptive.
+     * Determine the OP to use for this test based on the different test settings passed in.
+     * Then invoke the genericGoodEndToEndReAccessTest method to invoke the app, sleep and try to invoke the app again. On the second request, expect to land on the on the
+     * app again. Validate that the same tokens are used for each access - also make sure that there is NO re-authn message in the rp server log to show that we're using
+     * the same token
+     *
+     * @param tokenAutoRefresh - Flag indicating if the @OpenIdAuthenticationMechanismDefinition specifies that the token should be auto-refreshed
+     * @param providerAllowsRefresh - Flag indicating if the OP provider creates refresh_tokens
+     * @param notifyProvider - Flag indicating if the @OpenIdAuthenticationMechanismDefinition specifies that the provider should be notified
+     * @param idTokenExpiry - Flag indicating if the @OpenIdAuthenticationMechanismDefinition specifies that expiration of the id_token should be checked
+     * @param accessTokenExpiry - Flag indicating if the @OpenIdAuthenticationMechanismDefinition specifies that expiration of the access_token should be checked
+     * @throws Exception
+     */
+    public void genericGoodEndToEndReAccessTest(boolean tokenAutoRefresh, boolean providerAllowsRefresh, boolean notifyProvider, boolean idTokenExpiry,
+                                                boolean accessTokenExpiry) throws Exception {
+
+        String appName = getShortAppName(tokenAutoRefresh, providerAllowsRefresh, notifyProvider, idTokenExpiry, accessTokenExpiry);
+        String provider = determineProvider(providerAllowsRefresh, idTokenExpiry, accessTokenExpiry);
+
+        genericGoodEndToEndReAccessTest(appName, provider, providerAllowsRefresh);
+
+    }
+
+    /****************************************************************************************************************/
+    /* Tests */
+    /****************************************************************************************************************/
+
+    /**
+     * All Test will access a protected app and then wait for short lived tokens to expire.
+     */
+
+    /**
+     * This test will expect to use a refreshed token to access the protected app the second time. tokenAutoRefresh is true, the OP includes refresh tokens
+     * and both the id and access tokens have expired. The notifyProvider and the expiry settings should not matter in this case.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void BasicRefreshTests_tokenAutoRefreshTrue_providerAllowsRefreshTrue_notifyProviderTrue_idTokenExpiryTrue_accessTokenExpiryTrue() throws Exception {
+
+        genericGoodRefreshTest(true, true, true, true, true, TokenWasRefreshedSecondTime);
```

Comment:
```suggestion
        genericGoodRefreshTest(TokenAutoRefresh, true, NotifyProvider, IDTokenHonorExpiry, AccessTokenHonorExpiry, TokenWasRefreshedSecondTime);
```
Would make this a little clearer passing the constants you have defined so we don't get lost trying to figure out which boolean is supposed to mean what. Same goes for all calls.

Reply:
will update on the next delivery